### ISSUE TITLE
invalid-sequence-index when unpacking a sequence of sequences

### ISSUE THREAD

**Issue by Lalufu** [PyCQA/pylint#7424]
### Bug description

```
# pylint: disable=unused-variable,undefined-variable,invalid-name,missing-module-docstring,missing-function-docstring
def t():
    p1 = (
        "",
        24,
        64,
        {},
    )

    n = [
        p1,
        p1,
        p1,
        p1,
    ]

    for a1, a2, a3, a4 in n:
        i = a4["u_e"]
```

This seems related to the unpacking behaviour in the `for` loop. If `n` contains fewer entries than there are variables unpacked in the `for` loop, the behaviour does not occur (here, setting `n = [p1, p1 p1]` will make the error disappear.

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module t2
t2.py:18:12: E1126: Sequence index is not an int, slice, or instance with __index__ (invalid-sequence-index)
```
```


### Expected behavior

No error

### Pylint version

```shell
pylint 2.15.0
astroid 2.12.7
Python 3.8.13 (default, Apr 12 2022, 12:54:11) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

### Additional dependencies

astroid==2.12.7
dill==0.3.5.1
isort==5.10.1
lazy-object-proxy==1.7.1
mccabe==0.7.0
platformdirs==2.5.2
pylint==2.15.0
tomli==2.0.1
tomlkit==0.11.4
typing_extensions==4.3.0
wrapt==1.14.1
